### PR TITLE
TESTENG-1562: Call DocFX twice because DocFx has a caching issue for …

### PR DESCRIPTION
**Pull request by Sai-Nissankula**
…external code references

Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework.api/blob/master/CHANGELOG.md)

### CODE REVIEW

File: .github/workflows/ReleaseBranch.yaml [hunk 1]
```diff
@@ -277,6 +277,9 @@ jobs:
       run: |
         Invoke-WebRequest https://github.com/dotnet/docfx/releases/download/v2.58.4/docfx.zip -OutFile docfx.zip
         Expand-Archive -Path .\docfx.zip -DestinationPath DocFx
+        
+        # DocFx needs to be called two times because of a caching issue with external crefs. 
+        DocFx/docfx.exe source/Relativity.Testing.Framework.Api.Documentation/docfx.json
```
Comment (paulletang) [inline .github/workflows/ReleaseBranch.yaml:282]: There's not a way to clear the cache?

**Review by briantoms — APPROVED** [review]

File: .github/workflows/ReleaseBranch.yaml [hunk 2]
```diff
@@ -277,6 +277,9 @@ jobs:
       run: |
         Invoke-WebRequest https://github.com/dotnet/docfx/releases/download/v2.58.4/docfx.zip -OutFile docfx.zip
         Expand-Archive -Path .\docfx.zip -DestinationPath DocFx
+        
+        # DocFx needs to be called two times because of a caching issue with external crefs.
```
Comment (paulletang) [inline .github/workflows/ReleaseBranch.yaml:281]: Every time we get a VM, it is brand new so there shouldn't be anything cached.
Am I missing something happening before this?

Comment (Sai-Nissankula) [reply .github/workflows/ReleaseBranch.yaml:281]: I think the comment might be misleading. If docfx is called from a clean state, it breaks with warnings. So if I run git clean -fdx and then docfx, I see the warnings showing up on the pipeline. But if called a second time, the warnings go away.

File: .github/workflows/ReleaseBranch.yaml (hunk 1 above)
Comment (briantoms) [reply .github/workflows/ReleaseBranch.yaml:282]: The issue is the _lack_ of cache on the first run.

As best as I can tell, it seems like DocFX is divided into two steps, `metadata` and `build`. The xrefmap is only loaded in the `build` step, but it's needed during the `metadata` step to resolve the crefs in the XML comments. I can't find anything relevant in the DocFX documentation about loading the xrefmap sooner.

Comment (paulletang) [reply .github/workflows/ReleaseBranch.yaml:282]: @Sai-Nissankula the issue you raised with DocFX is for something other than this, right?
If we've looked through the docs, and there's nothing obvious for how to make this work with a single call, can you update the issue you made and/or make a new one for this?

Comment (Sai-Nissankula) [reply .github/workflows/ReleaseBranch.yaml:282]: @paulletang , yeah I was going to update it with this additional information once I confirm it works in production.

**Comment by Sai-Nissankula** [comment]
@paulletang , @briantoms , is there a way to force merge this? I am 100% sure this should not be impacting functional tests.

**Comment by paulletang** [comment]
> @paulletang , @briantoms , is there a way to force merge this? I am 100% sure this should not be impacting functional tests.

Yes, there is a way to force merge this, but it should be the exception, not the rule.
We should be keeping track of any tests that aren't behaving correctly so that we can address them, not just defaulting to force merge.

**Comment by paulletang** [comment]
> Yes, there is a way to force merge this, but it should be the exception, not the rule.
> We should be keeping track of any tests that aren't behaving correctly so that we can address them, not just defaulting to force merge.

To expand on this a bit more, admins can approve these types of merges, and I will help you do this, but please figure out which test is causing this issue first, create an issue for it, and provide some preliminary details for why it is failing.

**Comment by Sai-Nissankula** [comment]
I kicked off a second workflow to see if its reproducible, but another test failed with same error. I created an issue mentioning both tests: 
https://github.com/relativitydev/relativity.testing.framework.api/issues/194

**Comment by Sai-Nissankula** [comment]
@paulletang I've addressed your comments on the issue.

File: .github/workflows/ReleaseBranch.yaml (hunk 2 above)
Comment (paulletang) [reply .github/workflows/ReleaseBranch.yaml:281]: - [ ] Reword this comment to provide more clarity on the issue
- [ ] Link to the issue made with DocFX

**Review by briantoms — APPROVED** [review]

**Review by paulletang — APPROVED** [review]

**Pull request merged**